### ISSUE TITLE
/v1/jscript/variables?ext=.js&v=5.2.6 gives 404 

### ISSUE THREAD

**Issue by sephrat** [firefly-iii/firefly-iii#3399]
On my development system (Windows) I get a 404 which brakes most of Firefly III pages (no charts, some empty drop down lists, ...). 

I don't really know where to look at so I can try and fix the installation. 
There's nothing relevant in the Firefly III server log, 
PHP console says:
`[404]: GET /v1/jscript/accounts?ext=.js&v=5.2.6 - No such file or directory`
Browser console says:
`(index):820 GET http://localhost:8000/v1/jscript/variables?ext=.js&v=5.2.6 net::ERR_ABORTED 404 (Not Found)`

Any idea?

**Comment by JC5** [comment]
Mmmm that's very strange. Does the lin work without the stuff after the `?`?

**Comment by sephrat** [comment]
No, I still get a PHP 404 with http://localhost:8000/v1/jscript/variables.
http://localhost:8000/v1/js/app.js works fine.

I guess it has something to do with the dynamic routing, as it is not a static content in the public directory.
https://github.com/firefly-iii/firefly-iii/blob/09990acaa281521a66b3bae66996dd4aca3b16e5/routes/web.php#L600
I don't know how that works at all so I'm clueless.
While looking for solutions I found #550 and #830 but they're all solved by sorting out faulty Linux packages, which is not relevant with my Windows setup. I guess there's something wrong with my configuration somewhere. I use the PHP server embedded with Atom.

I don't really need it for now so I'll just leave it as it is. When I find time I'll try with a proper server setup like Apache or something.

**Comment by JC5** [comment]
The weird thing is this is a dynamic route just like all the others. If this one fails they should all fail, there's no difference.

**Comment by sephrat** [comment]
Yeah it's very strange.
![image](https://user-images.githubusercontent.com/34862846/82985772-e91c3d00-9ff4-11ea-8c74-ddf54adff287.png)

I tried with php-7.3.18 and php-7.4.6, same result. Same problem when I run the server from the command line directly, so I can eliminate a faulty Atom plugin.
Is there anything I should look at in php.ini? For reference: https://gist.github.com/sephrat/a61e0ff60906265a647b961a4c46fb87

**Comment by JC5** [comment]
Especially since all the URL's are fine. All I can imagine being the problem is the part after the question mark.

Since you know your way around the code, try editing the twig template where this file is being referenced (resources/views/v1/layout/default.twig). Remove everything after the ?.

The other JScript routes are used when you edit a reconciliation. Do those work?

**Comment by sephrat** [comment]
I'm not sure what you mean by editing a reconciliation, but I managed to reconcile an account and create a correction, edit said correction then delete it.

However some other stuff don't work like: 
- in /accounts/show/x I can't do mass actions (I get redirected to home screen)
- in /recurring/edit/x I can't pick a type of repetition, there is nothing in the dropdown list.
- charts are broken

I tried /v1/jscript/variables (dropped the parameters), still a 404. 
I also tried /v1/jscript/accounts and currencies, none of these work. 
Looks like anything under /v1/jscript is not routed, but everything else is. I tried putting a breakpoint in JavascriptController but it's not triggered. 

**Major breakthrough:** I put Firefly behind XAMPP and it just works! 
Not sure what's the difference between the 2 setups. They don't use exactly the same php versions (and .ini files) but I could not find anything fundamentally different.
